src/NetAnalyzers/CSharp/Microsoft.NetCore.Analyzers/Runtime/CSharpDetectPreviewFeatureAnalyzer.cs

```diff
@@ -58,5 +271,28 @@ private static bool TryGetPreviewInterfaceNodeForClassOrStructImplementingPrevie
             previewInterfaceNode = null;
             return false;
         }
+
+        private static bool IsSyntaxToken(SyntaxToken identifier, ISymbol previewInterfaceSymbol)
+        {
+            if (identifier.ValueText == previewInterfaceSymbol.Name)
+            {
+                return true;
+            }
+
+            return false;
+        }
```

Comment:
I'm not sure, but I feel this may not behave as expected.

Consider something like this:

```csharp
public class T { }

public class C<T>
{
    public void M1(global::T t) // Doesn't use the type parameter
    {
    }

    public void M2(T t) // Uses the type parameter.
    {
    }
}
```

Reply:
You're right. Maybe I should've made it more clear in the PR description. All the virtual methods here do Name comparisons, so we're only doing a best effort attempt to get the syntax node. I would really like to do `Symbol` comparisons, but I don't know how to do it efficiently :/ 

If someone has a suggestion on improving the logic here, I'm happy to implement it. @jmarolf @mavasani @sharwell

Reply:
@pgovind You may want to consider opening IOperation API proposal(s) in dotnet/roslyn if you didn't find the APIs you wished.